src/System.Runtime/tests/System/StringTests.cs

```diff
@@ -2714,17 +2748,21 @@ public static void ToUpper_TurkishI_TurkishCulture(string s, string expected)
             }, s.ToString(), expected.ToString()).Dispose();
         }
 
+        public static IEnumerable<object[]> ToUpper_TurkishI_EnglishUSCulture_MemberData() =>
+            ToUpper_TurkishI_MemberData(
+                new KeyValuePair<char, char>('\u0069', '\u0049'),
+                new KeyValuePair<char, char>('\u0130', '\u0130'),
+                new KeyValuePair<char, char>('\u0131', '\u0049'));
+
         [Theory]
-        [InlineData("H\u0069 World", "H\u0049 WORLD")]
```

Comment:
>"H\u0049 WORLD") [](start = 37, length = 16)

I would prefer having a string with Turkish I in the middle and not only the Turkish characters. This would be a real scenario rather than testing just one character.

Reply:
> I would prefer having a string with Turkish I in the middle and not only the Turkish characters. 

I don't understand.  What string do you want to see?  These flavors of strings already existed in the tests; I'm not adding them in from scratch.

Reply:
never mind, I didn't notice what you are doing in ToUpper_TurkishI_MemberData

I thought you were constructing the whole string from only Turkish I's but I was wrong.



---
In reply to: [178873636](https://github.com/dotnet/corefx/pull/28748#discussion_r178873636) [](ancestors = 178873636)